### PR TITLE
Add version to srf datasets

### PR DESCRIPTION
**Pull request by ekluzek**
Add a version attribute to surface datasets that can be checked for compatibility in ctsm.

Fixes #2723

- [x] Update the comments in the glacier code to remove the bit about it being because surface dataset isn't compatible.

### CODE REVIEW

File: src/main/surfrdMod.F90 [hunk 1]
```diff
@@ -45,6 +46,70 @@ module surfrdMod
...
+       ! For a few previous versions guess on the compatability version based on existence of variables
+       call check_var( ncid, 'PCT_OCEAN', exists)
+       if (exists) then
+         version = 5.2_r4
+       else
+         call check_var( ncid, 'PCT_CFT', exists)
+         if (exists) then
+            version = 5.0_r4
+         else
+            call check_var( ncid, 'PCT_GLC_MEC_GIC', exists)
```
Comment (ekluzek) [inline src/main/surfrdMod.F90]: This isn't quite right here.

File: tools/mksurfdata_esmf/src/mkfileMod.F90 [hunk 2]
```diff
@@ -126,6 +127,7 @@ subroutine mkfile_define_atts(pioid, dynlanduse)
     str = 'Community Land Model: CLM5'
     rcode = pio_put_att (pioid, pio_global, 'Source', trim(str))
     rcode = pio_put_att (pioid, pio_global, 'Version', trim(gitdescribe))
+    rcode = pio_put_att (pioid, pio_global, 'Dataset_Version', data_version)
```
Comment (ekluzek) [inline tools/mksurfdata_esmf/src/mkfileMod.F90:134]: The name should have something with compatibility or the like in the name.

File: tools/mksurfdata_esmf/src/mkfileMod.F90 [hunk 3]
```diff
@@ -105,6 +105,7 @@ subroutine mkfile_define_atts(pioid, dynlanduse)
     character(len= 10)   :: time               ! temporary
     character(len=  5)   :: zone               ! temporary
     integer              :: rcode
+    real(r4), parameter  :: data_version = 5.3   !  Compatibility version number of the datasets to create
```
Comment (ekluzek) [inline tools/mksurfdata_esmf/src/mkfileMod.F90:112]: Add comments that the version is only updated when changes to surface datasets make them incompatible with previous versions.

File: src/main/surfrdMod.F90 [hunk 4]
```diff
@@ -27,6 +27,7 @@ module surfrdMod
   save
   !
   ! !PUBLIC MEMBER FUNCTIONS:
+  public :: surfrd_check_file       ! Check that this surface dataset is compatible
```
Comment (ekluzek) [inline src/main/surfrdMod.F90]: Maybe:

surfrd_compatibility_check
surfrd_compat_check?

File: src/main/surfrdMod.F90 [hunk 5]
```diff
@@ -45,6 +46,70 @@ module surfrdMod
...
+      else if ( version > expected_version )then
+         description = 'newer'
+      end if
+      if ( masterproc )then
+         write(iulog,*) 'Input surface dataset is: ', trim(lfsurdat)
+         write(iulog,'(3a)') 'This surface dataset is ', trim(description), ' and incompatable with this version of CTSM'
+         write(iulog,'(a,f3.1,a,f3.1)') 'Dataset version = ', version, ' Version expected = ', expected_version
+         write(iulog,*) errMsg(sourcefile, __LINE__)
+      end if
+      call endrun(msg="ERROR: Incompatble surface dataset")
```
Comment (ekluzek) [inline src/main/surfrdMod.F90]: Fix spelling.

File: tools/mksurfdata_esmf/src/mkfileMod.F90 (hunk 2 above)
Comment (ekluzek) [reply tools/mksurfdata_esmf/src/mkfileMod.F90:134]: In meeting with @slevis-lmwg he suggests we don't need to add compatibility to the name.

**Review by slevis-lmwg — APPROVED** [review]
@ekluzek and I looked this over, and I'm approving.

File: src/main/surfrdMod.F90 (hunk 4 above)
Comment (ekluzek) [reply src/main/surfrdMod.F90]: latter

File: src/main/surfrdMod.F90 (hunk 1 above)
Comment (ekluzek) [reply src/main/surfrdMod.F90]: Corrected

**Pull request merged**